node/node.go

```diff
@@ -120,45 +127,79 @@ func (nn *nexNode) Validate() error {
 	return errs
 }
 
-func (nn *nexNode) Start() {
-	nn.initializeSupervisionTree()
+func (nn *nexNode) Start() error {
+	return nn.initializeSupervisionTree()
 }
 
-func (nn *nexNode) initializeSupervisionTree() {
+func (nn *nexNode) initializeSupervisionTree() error {
 	var options gen.NodeOptions
 
+	nodeID, err := nn.publicKey.PublicKey()
+	if err != nil {
+		fmt.Printf("Unable to start node '%s': %s\n", nn.name, err)
+		return err
+	}
+
 	// create applications that must be started
-	apps := []gen.ApplicationBehavior{
-		observer.CreateApp(observer.Options{}), // TODO: opt out of this via config
-		actors.CreateNodeApp(*nn.options),      // copy options
+	options.Applications = []gen.ApplicationBehavior{
+		observer.CreateApp(observer.Options{}),           // TODO: opt out of this via config
+		actors.CreateNodeApp(nodeID, nn.nc, *nn.options), // copy options
 	}
-	options.Applications = apps
 
 	// disable default logger to get rid of multiple logging to the os.Stdout
 	options.Log.DefaultLogger.Disable = true
 
 	// https://docs.ergo.services/basics/logging#process-logger
 	// https://docs.ergo.services/tools/observer#log-process-page
 
-	nodeName := "nex@localhost"
-
 	// starting node
-	node, err := ergo.StartNode(gen.Atom(nodeName), options)
+	node, err := ergo.StartNode(gen.Atom(nn.name), options)
```

Comment:
I specifically indicated that this needs to be a constant that is not mutable or configurable.

Reply:
It is not configurable: https://github.com/synadia-io/nex/blob/feat/vnext-node/node/node.go#L42

This seemed like a good compromise since we log the name quite a bit.

Reply:
ok. I saw the `default` prefix and assumed there was a way to override it. Let's change the constant to something like `ergoNodeName` so that it's clear it's not actually related to the "nex node".

Reply:
We also shouldn't be logging that name at all. It's not useful for anything. (Unless it's coming "for free" with the actor logging, in which case we can maybe tweak that so we don't log the node name in another PR)